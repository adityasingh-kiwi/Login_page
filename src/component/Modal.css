.form{
    background-color:#2b3e50;
}

.user {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    text-align: left;
    width: 400px;
    line-height: 26px;
    margin-bottom: 10px;
  }