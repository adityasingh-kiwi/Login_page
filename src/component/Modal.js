import React, { useEffect, useState } from 'react';
import 'antd/dist/antd.css';
import { getTokens, onMessageListener } from "../firebase";
import firebase from "../firebase";
import { Modal, Button } from 'antd';
import './Modal.css';
import RequestService from "./api";
import { getToken } from '@firebase/messaging';
// import { getMessaging,getToken  } from "firebase/messaging";


const ModalOne = ({ isopen, isOpenClose, getdata }) => {
    const [isModalVisible, setIsModalVisible] = useState(false);
    const [dataModal, setDatamodal] = useState({
        firstName: "",
        lastName: "",
        userEmail: ""

    })

    // const loader = async () => {
    //     console.log("getToken-------------->>",getToken);
    //     await getToken().then((webPushTokenFCM) =>{
    //         console.log("webpushhhh--------->>",webPushTokenFCM);

    //     })
    // }      
    // useEffect(()=>{

    //     loader();

    //     // const msg= firebase.messaging();

    //     //  msg.requestPermission().then(()=>{
    //     //       return msg.getTokens()
    //     //  }).then((data)=>{
    //     //         console.log("token",data)
    //     // })
    // },[])


    // useEffect(() => {
    //     let modifiedObject = { ...dataModal }
    //     modifiedObject["userEmail"] = getdata.email;
    //     setDatamodal(modifiedObject);

    // }, [])

    const handleModalchange = (e) => {

        setDatamodal({ ...dataModal, [e.target.name]: e.target.value })
    }


    const showModal = () => {
        setIsModalVisible(true);
    };

    const handleOk = async () => {

        try {
            let userPassword = ""

            const { firstName,
                secondName,
                userEmail } = dataModal;
            console.log("getdata.id---------------1111->>>", getdata.id);
            let response = await RequestService.putOne(getdata.id, {
                firstName,
                secondName,
                userEmail,
                userPassword
            })
            console.log("response----------------------->>>,", response);
        } catch (error) {

        }
        isOpenClose(false);

    };

    const handleCancel = () => {
        isOpenClose(false);
    };



    
    return (
        <>

            <Modal title="Update the User Details" visible={isopen} onOk={handleOk} onCancel={handleCancel}>
                <div className="form">
                    <form >
                        <div>
                            First Name:<input type="text" name="firstName" className="user" type="text" value={dataModal.firstName} onChange={handleModalchange} placeholder="firstname" required />

                            Last Name :<input type="text" name="lastName" className="user" type="text" value={dataModal.lastName} onChange={handleModalchange} placeholder="Lastname" required />
                        </div>
                        <div>
                            Email id:<input type="Email" name="userEmail" className="user" type="email" value={dataModal.userEmail} onChange={handleModalchange} placeholder="email id" required />
                        </div>
                        <div>
                            <span>
                                Gender: <input type="radio" name="gender" value="male" /> Male
                                <input type="radio" name="gender" value="female" /> Female<br />
                                <input type="radio" name="gender" value="other" /> Other
                            </span>
                        </div>
                        <div>
                            Upload photo: <input type="file" name="fileupload" accept="image/*" />
                        </div>




                    </form>
                </div>
            </Modal>
        </>
    );
};

export default ModalOne;