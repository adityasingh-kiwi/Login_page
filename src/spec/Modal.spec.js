import Modal from '../component/Modal';
import { mount, render, shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ExceptionMap } from 'antd/lib/result';

configure({ adapter: new Adapter() });

describe('Modal_test',()=>{

   it('input_check_firstname',()=>{
    const wrapper = shallow(<Modal />);
    const input = wrapper.find(".userFirstName");
    input.simulate("change", { target: { value: "Aditya" } });


   })
   it('input_check_firstname and same value',()=>{
    const wrapper = shallow(<Modal />);
    const input = wrapper.find('#firstName_Modal');
    expect(input.exists('#firstName_Modal')).toEqual(true);

    input.find('#firstName_Modal').simulate('change', {
      target: {
        name:"firstName",
        value: 'Aditya',
      },
    }); 

    expect(input.find('#firstName_Modal').prop('value')).toEqual("");


   })
   it('input_check_Lastname',()=>{
    const wrapper = shallow(<Modal />);
    const input = wrapper.find(".userLastName");
    input.simulate("change", { target: { value: "Singh" } });
   })
   it('input_check_Email',()=>{
    const wrapper = shallow(<Modal />);
    const input = wrapper.find(".userEmail");
    input.simulate("change", { target: { value: "adityasingh@gmail.com" } });
   })
   it('input_check_Gender',()=>{
    const wrapper = shallow(<Modal />);
    const input = wrapper.find(".gender"); 
  
    input.at(0).simulate('change', { target: { checked: true } });
   })


it('Snapshot_modal',()=>{
    let wrapper=mount(<Modal/>)
    expect(wrapper).toMatchSnapshot();
    

  })
  
  it('FirstName_handle',()=>{
    let wrapper=shallow(<Modal/>)
     expect(wrapper.exists('#firstName_Modal')).toEqual(true);
     wrapper.find('#firstName_Modal').simulate('change', {
     target: {
       name:"firstName",
       value: 'Aditya',
     },
   })
 expect(
  wrapper.find('#firstName_Modal').prop('defaultValue'),
  ).toBeTruthy();


})
it('LastName_handle',()=>{
  let wrapper=shallow(<Modal/>)
   wrapper.find('#user_lastName').simulate('change', {
   target: {
     name:"lastName",
     value: 'Singh',
   },
 })
expect(
wrapper.find('#user_lastName').prop('defaultValue'),
).toBeTruthy();


})
it('Email_handle',()=>{
  let wrapper=shallow(<Modal/>)
   wrapper.find('#user_Email').simulate('change', {
   target: {
     name:"userEmail",
     value: 'Singh',
   },
 })
expect(
wrapper.find('#user_Email').prop('defaultValue'),
).toBeTruthy();


})
})

