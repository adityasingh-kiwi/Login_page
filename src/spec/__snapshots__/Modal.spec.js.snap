// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal_test Snapshot_modal 1`] = `
<ModalOne>
  <Modal
    confirmLoading={false}
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Update the User Details"
    visible={false}
    width={520}
  >
    <Dialog
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <ForwardRef(CloseOutlined)
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      focusTriggerAfterClose={true}
      footer={
        <LocaleReceiver
          componentName="Modal"
          defaultLocale={
            Object {
              "cancelText": "Cancel",
              "justOkText": "OK",
              "okText": "OK",
            }
          }
        >
          [Function]
        </LocaleReceiver>
      }
      maskTransitionName="ant-fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      onOk={[Function]}
      prefixCls="ant-modal"
      title="Update the User Details"
      transitionName="ant-zoom"
      visible={false}
      width={520}
      wrapClassName=""
    >
      <PortalWrapper
        visible={false}
      />
    </Dialog>
  </Modal>
</ModalOne>
`;
